lib/restforce/sobject.rb

```diff
@@ -63,5 +67,10 @@ def ensure_id
       return true if self.Id?
       raise ArgumentError, 'You need to query the Id for the record first.'
     end
+
+    def ensure_write_access
+      return true unless Restforce.configuration.read_only
+      raise Restforce::UnauthorizedError, 'You are in read only mode.'
```

Comment:
Might be nice to make a new error class for this to avoid overloaded `UnauthorizedError`.